main.go

```diff
@@ -420,6 +485,41 @@ func postPost(c *gin.Context, d database.IDatabase) {
 	}
 }
 
+func editPost(c *gin.Context, d database.IDatabase, postId string) {
+	var post model.Post
+	c.BindJSON(&post)
+	err := d.EditPost(&post)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+	} else {
+		post, err = d.GetPost(postId)
+		if err != nil {
+			log.Error("Cannot get post")
+		}
+
+		bytes, err := json.Marshal(&post)
+		if err != nil {
+			return
+		}
+
+		c.JSON(http.StatusOK, post)
+		if c, err := gRedisConn(); err != nil {
+			log.Printf("Error on redis conn. %s", err)
+		} else {
+			c.Do("PUBLISH", "posts", bytes)
```

Comment:
Sounds good. I need to educate myself on Reddis a bit, I mostly copied/pasted this from a different function.

Reply:
You can think of it as just a PUB/SUB model, where if you propagate something to it, it will publish that event to subscribers. I publish out the posts so that the front end can add new ones as they show up, which makes the board real time (which is cool).